modules/nextflow/src/main/groovy/nextflow/executor/BashWrapperBuilder.groovy

```diff
@@ -250,7 +250,7 @@ class BashWrapperBuilder {
         binding.helpers_script = getHelpersScript()
 
         if( runWithContainer ) {
-            binding.container_boxid = 'export NXF_BOXID="nxf-$(dd bs=18 count=1 if=/dev/urandom 2>/dev/null | base64 | tr +/ 0A)"'
+            binding.container_boxid = 'export NXF_BOXID="nxf-$(dd bs=18 count=1 if=/dev/urandom 2>/dev/null | base64 | tr +/ 0A | tr -d \'\\r\\n\')"'
```

Comment:
I think `tr +/ 0A` is already stripping `\n`, ideally the stripping of `\r` should be added to the first `tr`

Reply:
I think `tr +/ 0A` means replace `+` with `0` and `/` with `A`, though not sure why it's doing that. You have to do `tr -d` to delete characters.

Reply:
ok, i had  misinterpreted `0A` as the hex for `\n`